### ISSUE TITLE
TR LegIntermediate doesnt conform to XSD schema

### ISSUE THREAD

**Issue by vasile** [openTdataCH/ojp-sdk#42]
Given any TR Bern - Zürich or with `Params/IncludeIntermediateStops` = True 

This `LegIntermediate` node is returned and doesnt conform with the schema 
https://vdvde.github.io/OJP/develop/index.html#LegIntermediateStructure
- `ServiceArrival` and `ServiceDeparture`  are required

```
  <LegIntermediate>
      <siri:StopPointRef>132</siri:StopPointRef>
      <StopPointName>
          <Text xml:lang="de">Bahn-2000-Strecke</Text>
      </StopPointName>
      <Order>2</Order>
  </LegIntermediate>
```

Expected:
- Arrival / Departure to be present (it is in HRDF for these special stops) 
- `NoBoardingAtStop` + `NoAlightingAtStop` nodes should be present

**Comment by vasile** [comment]
~The issue is labelled as `blocker` due to required ServiceArrival/ServiceDeparture in the TR model~
https://github.com/openTdataCH/ojp-ios/blob/main/Sources/OJP/Models/OJPv2%2BTrip.swift#L275-L276

But for now we can make these optional until backend fix is present

**Comment by r3to** [comment]
@TO-mdv is this a complex bug to be fixed? 
If not, it would be cool to prfioritize this, so we don't have to implement workarounds.

**Comment by TO-mdv** [comment]
@r3to it seems to be a data problem. I can not reproduce it with my requests. Example S26 between Zürich - Bern delivers the elements:
`<LegIntermediate>
    <siri:StopPointRef>ch:1:sloid:2113:3:5</siri:StopPointRef>
    <StopPointName>
        <Text xml:lang="de">Aarau</Text>
    </StopPointName>
    <NameSuffix>
        <Text xml:lang="de">NO_DATA</Text>
    </NameSuffix>
    <PlannedQuay>
        <Text xml:lang="de">5</Text>
    </PlannedQuay>
    <EstimatedQuay>
        <Text xml:lang="de">5</Text>
    </EstimatedQuay>
    <ServiceArrival>
        <TimetabledTime>2024-07-01T07:07:00Z</TimetabledTime>
        <EstimatedTime>2024-07-01T07:07:00Z</EstimatedTime>
    </ServiceArrival>
    <ServiceDeparture>
        <TimetabledTime>2024-07-01T07:08:00Z</TimetabledTime>
        <EstimatedTime>2024-07-01T07:08:12Z</EstimatedTime>
    </ServiceDeparture>
    <Order>2</Order>
</LegIntermediate>`

Same for the NoBoardingAtStop + NoAlightingAtStop, example Zürich - St. Margrethen SG delivers the attribut:
`<LegIntermediate>
    <siri:StopPointRef>ch:1:sloid:6000:3:5</siri:StopPointRef>
    <StopPointName>
        <Text xml:lang="de">Winterthur</Text>
    </StopPointName>
    <NameSuffix>
        <Text xml:lang="de">NO_DATA</Text>
    </NameSuffix>
    <PlannedQuay>
        <Text xml:lang="de">5</Text>
    </PlannedQuay>
    <EstimatedQuay>
        <Text xml:lang="de">5</Text>
    </EstimatedQuay>
    <ServiceArrival>
        <TimetabledTime>2024-12-13T12:57:00Z</TimetabledTime>
    </ServiceArrival>
    <ServiceDeparture>
        <TimetabledTime>2024-12-13T12:58:00Z</TimetabledTime>
    </ServiceDeparture>
    <Order>3</Order>
    <NoAlightingAtStop>true</NoAlightingAtStop>
</LegIntermediate>`

**Comment by TO-mdv** [comment]
@r3to I cannot reproduce this issue. Example TR S26 between Zürich-Bern delivers ServiceArrival and ServiceDeparture :
```
<LegIntermediate>
    <siri:StopPointRef>ch:1:sloid:2113:3:5</siri:StopPointRef>
    <StopPointName>
        <Text xml:lang="de">Aarau</Text>
    </StopPointName>
    <NameSuffix>
        <Text xml:lang="de">NO_DATA</Text>
    </NameSuffix>
    <PlannedQuay>
        <Text xml:lang="de">5</Text>
    </PlannedQuay>
    <EstimatedQuay>
        <Text xml:lang="de">5</Text>
    </EstimatedQuay>
    <ServiceArrival>
        <TimetabledTime>2024-07-01T07:07:00Z</TimetabledTime>
        <EstimatedTime>2024-07-01T07:07:00Z</EstimatedTime>
    </ServiceArrival>
    <ServiceDeparture>
        <TimetabledTime>2024-07-01T07:08:00Z</TimetabledTime>
        <EstimatedTime>2024-07-01T07:08:12Z</EstimatedTime>
    </ServiceDeparture>
    <Order>2</Order>
</LegIntermediate>
```

**Comment by TO-mdv** [comment]
Also the NoBoardingAtStop + NoAlightingAtStop are delivered, example Zürich - St Margrethen SG:

```
<LegIntermediate>
    <siri:StopPointRef>ch:1:sloid:6000:3:5</siri:StopPointRef>
    <StopPointName>
        <Text xml:lang="de">Winterthur</Text>
    </StopPointName>
    <NameSuffix>
        <Text xml:lang="de">NO_DATA</Text>
    </NameSuffix>
    <PlannedQuay>
        <Text xml:lang="de">5</Text>
    </PlannedQuay>
    <EstimatedQuay>
        <Text xml:lang="de">5</Text>
    </EstimatedQuay>
    <ServiceArrival>
        <TimetabledTime>2024-12-13T12:57:00Z</TimetabledTime>
    </ServiceArrival>
    <ServiceDeparture>
        <TimetabledTime>2024-12-13T12:58:00Z</TimetabledTime>
    </ServiceDeparture>
    <Order>3</Order>
    <NoAlightingAtStop>true</NoAlightingAtStop>
</LegIntermediate>
```

**Comment by r3to** [comment]
@TO-mdv The "Aarau" example is from a connection with an actual stop.

What @vasile refered to, were connections like the IC8 where "Bahn-200-Strecke" is shown as `LegIntermediate` but without the other required elements. (Note: this snippet is from Prod, as the other environments are not reachable at the moment)

```xml
<LegIntermediate>
  <siri:StopPointRef>132</siri:StopPointRef>
  <StopPointName>
    <Text xml:lang="de">Bahn-2000-Strecke</Text>
  </StopPointName>
  <Order>2</Order>
</LegIntermediate>
```

**Comment by TO-mdv** [comment]
@r3to Ah perfect, now I've got the example. I will create a bug for this, hopefully it's a small issue, so we will fix it soon

**Comment by TO-mdv** [comment]
Wir werden Durchfahrten als kurzfristige Lösung komplett aus der Antwort löschen. Hintergrund: wenn wir diese mit den An-/Abfahrtszeiten übernehmen, wir von der Echtzeit jedoch diese Durchfahrten nicht erhalten, werden diese immer als "Zusatzhalte" interpretiert.
[SH-3219](https://youtrack.mentz.net/issue/SH-3219) OJP2.0 TR Durchfahrtshaltestellen nicht ausgeben (Bahn-2000-Strecke)

**Comment by TO-mdv** [comment]
Released on INT, Release on PROD 25.7.2024

The response will for the moment not contain the "Durchfahrtshaltestellen" anymore, Matthias Günter will check how we proceed with this stops. Examples:

- Bahn-2000-Strecke
- Gotthard-Basistunnel
- Gotthard-Panoramastrecke
- ...
